dds_web/security/project_keys.py

```diff
@@ -0,0 +1,146 @@
+""" Code for generating project related keys """
+import os
+
+import argon2
+import cryptography
+import flask
+import gc
+
+import dds_web
+from dds_web.errors import KeyNotFoundError
+
+
+def derive_key(user, password):
+    """Derive a KEK from users password."""
+    # Salt used for key derivation
+    if user.kd_salt is None:
+        exception = Exception("Access to project data is not properly setup for the user!")
+        flask.current_app.logger.exception(exception)
+        raise exception
+
+    # Derive the key
+    derived_key = argon2.low_level.hash_secret_raw(
+        secret=password.encode(),
+        salt=user.kd_salt,
+        time_cost=2,
+        # memory_cost=4194304, # This makes it crash locally
+        memory_cost=1000,  # Just changed to something so that it wouldn't crash
```

Comment:
I recommend we create an issue that is for setting a config for development purposes. In that config, we could set a small value for the key derivation memory cost, disable hotp or other features that we don't use or use in a limited way on a day to day basis. If this breaks the build, perhaps we should empower our build environment, if possible.

Reply:
https://github.com/ScilifelabDataCentre/dds_web/issues/812#issue-1124288020

Reply:
@talavis Could you set a config value for development regarding the `memory_cost` mentioned here?

Reply:
Just to mention: `memory cost` is in kibibytes, so `memory_cost=4194304`means that you want to use 4GiB of RAM. In the case of my docker setup, that's beyond my resource limits.

Reply:
It is 4GiB of RAM. I too had the issue of running out of memory, but changed computer so no issue for me right now. I followed the documentation and the help of a calculator. How would you write it?